python/lsst/meas/algorithms/objectSizeStarSelector.py

```diff
@@ -430,21 +428,35 @@ def selectStars(self, exposure, sourceCat, matches=None):
         centers, clusterId = _kcenters(width, nCluster=4, useMedian=True,
                                        widthStdAllowed=self.config.widthStdAllowed)
 
-        if display and plotMagSize and pyplot:
-            fig = plot(mag, width, centers, clusterId,
-                       magType=self.config.sourceFluxField.split(".")[-1].title(),
-                       marker="+", markersize=3, markeredgewidth=None, ltype=':', clear=True)
-        else:
-            fig = None
-
         clusterId = _improveCluster(width, centers, clusterId,
                                     nsigma=self.config.nSigmaClip,
                                     widthStdAllowed=self.config.widthStdAllowed)
 
-        if display and plotMagSize and pyplot:
-            plot(mag, width, centers, clusterId, marker="x", markersize=3, markeredgewidth=None, clear=False)
+        if display and plotMagSize:
+            self._plot(display, displayExposure, exposure, plotMagSize, mag, width,
+                       centers, clusterId, sourceCat, good)
 
         stellar = (clusterId == 0)
+        starCat = SourceCatalog(sourceCat.table)
+        goodSources = [s for g, s in zip(good, sourceCat) if g]
+        for isStellar, source in zip(stellar, goodSources):
+            if isStellar:
+                starCat.append(source)
+
+        return Struct(
+            starCat=starCat,
+        )
+
+    def _plot(self, display, displayExposure, exposure, plotMagSize, mag, width,
```

Comment:
Making this a separate method is probably wise (though I'm sorry all that event handling is needed; it seems far too details for diagnostic plots, and like it belongs in the display library, not here). However, I find the design confusing in that:
- The main interface for plotting is to call to the private `_plot` method, which then calls a public `plot` function. It feels backwards to have to call private code which calls public code; usually things go the other way -- a public function will call a private function for some implementation. Some renaming would help.
- Please add a doc string to the `_plot` method. It's not clear to me what some of the arguments are.